### PR TITLE
Making attribute names more precise: obsdim->obs_dimname, timedim->time_varname

### PR DESCRIPTION
**Pull request by knutfrode**
So far mainly a mechanical search & replace
Thus some comments are still inconsistent, e.g. refering to `time_varname` as time dimension.

Also remains to add an attribute `trajectory_dimname`, but can perhaps be made in a subsequent pull request. So far this is hardcoded as "trajectory" at some places.

### CODE REVIEW

**Comment by gauteh** [comment]
Should we use `dim` and not `dimname` in the public API? I think that is what xarray is doing: https://docs.xarray.dev/en/latest/generated/xarray.Dataset.reduce.html#xarray.Dataset.reduce

**Comment by knutfrode** [comment]
Hm, ok yes, we should aim to be consistent with Xarray.
Shall I then replace all `obs_dimname` with `obs_dim` ?

But `time_varname` I guess is ok, as `time_var` could be interpreted both as the name of the time variable or as the variable itself. For dimension we do not have this dual possibility.

**Comment by gauteh** [comment]
Yes, I think this is the best compromise?

File: trajan/accessor.py [hunk 1]
```diff
@@ -90,63 +90,63 @@ def __new__(cls, ds):
...
 
-        elif len(ds[timedim].shape) == 2:
+        elif len(ds[time_varname].shape) == 2:
             logger.debug('Detected un-structured (2D) trajectory dataset')
             ocls = Traj2d
 
         else:
             raise ValueError(
-                f'Time dimension has shape greater than 2: {ds["timedim"].shape}'
+                f'Time dimension has shape greater than 2: {ds["time_varname"].shape}'
```
Comment (knutfrode) [inline trajan/accessor.py:149]: This an example of why old names were not precise enough.
Thus `Time dimension has shape greater than 2` should now be changed to the more precise `Time variable has more than 2 dimensions`

Comment (gauteh) [reply trajan/accessor.py:149]: Yes, also this probably needs to be: ds[self.time_varname] without the internal quotes.

**Pull request merged**